services-camel/src/main/java/edu/unc/lib/cdr/ReplicationProcessor.java

```diff
@@ -0,0 +1,218 @@
+		for (String replicationPath : replicationPaths) {
+			if (!Files.exists(Paths.get(replicationPath))) {
+				throw new ReplicationDestinationUnavailableException(String.format("Unable to find replication destination %s", replicationPath));
+			}
+		}
+		
+		return true;
+	}
+	
+	private String createFilePath(String basePath, String originalFileChecksum) {
+		String[] tokens = Iterables.toArray
+				(Splitter.fixedLength(2).split( originalFileChecksum), 
+						String.class);
+		
+		String remotePath = new StringJoiner("/")
+			.add(basePath)
+			.add(tokens[0])
+			.add(tokens[1])
+			.add(tokens[2])
+			.toString();
+		
+		return remotePath;
+	}
+	
+	private String createRemoteSubDirectory(String baseDirectory, String binaryChecksum) {
+		String replicationPath = createFilePath(baseDirectory, binaryChecksum);
+		Path fullPath = Paths.get(replicationPath);
+		
+		if (!Files.exists(fullPath)) {
+			new File(replicationPath).mkdirs();
+		}
+
+		return replicationPath;
+	}
+	
+	private String dbFilePath(String uri, String BinaryChecksum) {
+		String filePath = "/tmp/" + BinaryChecksum;
+		InputStream response = null;
+		
+		try {
+			URLConnection connection = new URL(uri).openConnection();
+			connection.setRequestProperty("Accept-Charset", UTF_8.toString());
+			response = connection.getInputStream();
+			
+			byte[] buffer = new byte[response.available()];
+			response.read(buffer);
+		 
+			File targetFile = new File(filePath);
+			try(OutputStream outStream = new FileOutputStream(targetFile)) {
+				outStream.write(buffer);
+			}
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		
+		return filePath;
+	}
+	
+	private String getFileChecksum(String filePath) {
+		String checksum = null;
+		try {
+			checksum = DigestUtils.sha1Hex(new FileInputStream(filePath));
+		} catch (IOException e) {
+			throw new ReplicationException(String.format("Unable to compute checksum for %s", filePath), e);
+		}
+		
+		return checksum;
+	}
+	
+	private void verifyChecksums(String originalFileChecksum, String replicatedFilePath) {
+		String remoteChecksum = getFileChecksum(replicatedFilePath);
+
+		if (originalFileChecksum.equals(remoteChecksum)) {
+			log.info("Local and remote checksums match for {}", replicatedFilePath);
+		} else {
+			throw new ReplicationException(String.format("Local and remote checksums did not match %s %s", originalFileChecksum, remoteChecksum));
+		}
+	}
+	
+	private void replicate(String binaryPath, String originalFileChecksum, String[] replicationLocations, String binaryMimeType, String binaryUri) throws InterruptedException {
+		String remoteFile = null;
+		String localBinary = null;
+		
+		try {
+			for (String location : replicationLocations) {
+				String fullPath = createRemoteSubDirectory(location, originalFileChecksum);
+				
+				if (Files.exists(Paths.get(binaryPath))) {
+					localBinary = binaryPath;
+				} else {
+					localBinary = dbFilePath(binaryUri, originalFileChecksum);
```

Comment:
The temp file from this is going to need to be cleaned up once it is finished being replicated, which we absolutely don't want to do for the other case.